### PR TITLE
[tbb] Set fatal build tool requirements for UNIX

### PR DESCRIPTION
**Pull request by JackBoosY**
Related: #7727.

### CODE REVIEW

**Review by PhoebeHui — APPROVED** [review]

File: ports/tbb/terminate-when-buildtool-notfound.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
+diff --git a/cmake/TBBBuild.cmake b/cmake/TBBBuild.cmake
+index a2222e3..a208258 100644
+--- a/cmake/TBBBuild.cmake
++++ b/cmake/TBBBuild.cmake
+@@ -154,11 +154,11 @@ function(tbb_build)
+         set(make_tool_name ndk-build)
+     endif()
+ 
+-    find_program(TBB_MAKE_TOOL ${make_tool_name} DOC "Make-tool to build Intel TBB.")
++    find_program(TBB_MAKE_TOOL ${make_tool_name} DOC "Make-tool to build Intel TBB." REQUIRED)
```
Comment (cbezault) [inline ports/tbb/terminate-when-buildtool-notfound.patch]: Is setting this `REQUIRED` necessary with the `FATAL_ERROR` below?

File: ports/tbb/terminate-when-buildtool-notfound.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+--- a/cmake/TBBBuild.cmake
++++ b/cmake/TBBBuild.cmake
+@@ -158,7 +158,7 @@ function(tbb_build)
+     mark_as_advanced(TBB_MAKE_TOOL)
+ 
+     if (NOT TBB_MAKE_TOOL)
+-        message(STATUS "Intel TBB can not be built: required make-tool (${make_tool_name}) was not found")
++        message(FATAL_ERROR "Intel TBB can not be built: required make-tool (${make_tool_name}) was not found")
+         set(${tbb_build_CONFIG_DIR} ${tbb_build_CONFIG_DIR}-NOTFOUND PARENT_SCOPE)
+         return()
```
Comment (cbezault) [inline ports/tbb/terminate-when-buildtool-notfound.patch]: Is there any obvious reason it tries to continue on?

Comment (JackBoosY) [reply ports/tbb/terminate-when-buildtool-notfound.patch]: Well, I think `message(FATAL_ERROR)` will terminate the build process.

**Review by dan-shaw — APPROVED** [review]

**Comment by dan-shaw** [comment]
LGTM

**Pull request merged**